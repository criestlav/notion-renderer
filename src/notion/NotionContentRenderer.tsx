import React from 'react';
import TextBlock from './TextBlock';
import BlankBlock from './BlankBlock';
import BulletedList from './BulletedList';
import CodeBlock from './CodeBlock';
import { HeaderBlock, SubHeaderBlock, SubSubHeaderBlock } from './HeadingBlock';
import ImageBlock from './ImageBlock';

interface NotionContentRendererProps {
    recordMap: any;
    id: string;
    imageSource: (url: string) => string;
}

const NotionContentRenderer = (props: NotionContentRendererProps) => {
    const { recordMap, id } = props;
    const content = recordMap.block[`${id}`].value.content;
    const items = content.map((item: string) => {
        const data = recordMap.block[`${item}`];
        if (data) {
            return data.value;
        }
    });

    const renderedItems = items.map((item: any, index: number) => {
        if (!item) {
            return undefined;
        }
        switch (item.type) {
            case 'bulleted_list':
                return (
                    <BulletedList properties={item.properties} key={index} />
                );
            case 'code':
                return <CodeBlock properties={item.properties} key={index} />;
            case 'divider':
                return <hr key={index} />;
            case 'header':
                return <HeaderBlock properties={item.properties} key={index} />;
            case 'sub_header':
                return (
                    <SubHeaderBlock properties={item.properties} key={index} />
                );
            case 'sub_sub_header':
                return (
                    <SubSubHeaderBlock
                        properties={item.properties}
                        key={index}
                    />
                );
            case 'text':
                return <TextBlock properties={item.properties} key={index} />;
            case 'image':
                return (
                    <ImageBlock
                        id={item.id}
                        properties={item.properties}
                        format={item.format}
                        imageSource={props.imageSource}
                        key={index}
                    />
                );
            default:
                return <BlankBlock key={index} />;
        }
    });

    return <div>{renderedItems}</div>;
};

export default NotionContentRenderer;
