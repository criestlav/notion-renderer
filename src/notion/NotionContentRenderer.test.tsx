import * as fs from 'fs';
import * as path from 'path';
import renderer from 'react-test-renderer';
import React from 'react';
import NotionContentRenderer from './NotionContentRenderer';

test('content is rendered', async () => {
    const testDataFile = path.resolve(
        __dirname,
        '../../notion-renderer.data.json'
    );
    const testData = JSON.parse(fs.readFileSync(testDataFile).toString());
    const { id, recordMap } = testData;
    const imageSource = (url: string) => url;
    const component = renderer.create(
        <NotionContentRenderer
            recordMap={recordMap}
            id={id}
            imageSource={imageSource}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
