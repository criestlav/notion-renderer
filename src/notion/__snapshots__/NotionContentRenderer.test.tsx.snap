// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content is rendered 1`] = `
<div>
  <h3>
    header
  </h3>
  <h4>
    subheader
  </h4>
  <h5>
    sub sub header
  </h5>
  <span>
    Â 
  </span>
</div>
`;
